contracts/connection-router/src/msg.rs

```diff
@@ -7,6 +7,8 @@ pub struct InstantiateMsg {
     pub admin_address: String,
     // governance votes on chains being added or upgraded
     pub governance_address: String,
+    // the address of the nexus gateway
+    pub nexus_gateway: String,
```

Comment:
If the router needs to know the address of the nexus gateway, and the nexus gateway needs to know the address of the router, how do we instantiate the two contracts?

Reply:
probably better to register the nexus gateway after instantiation?

Reply:
yeah I think that's best. I'm not sure the best way to do this though. The two ways I can think of is to either add a new execute msg called `RegisterNexusGateway`, or to just handle it inside `RegisterChain`. I don't really want to add a new execute msg to the API, but I also don't want to complicate the implementation of `RegisterChain`.